lib/ansible/module_utils/redfish_utils.py

```diff
@@ -697,54 +731,176 @@ def list_users(self):
         result["entries"] = users_results
         return result
 
-    def add_user(self, user):
-        uri = self.root_uri + self.accounts_uri + "/" + user['userid']
-        username = {'UserName': user['username']}
-        pswd = {'Password': user['userpswd']}
-        roleid = {'RoleId': user['userrole']}
-        enabled = {'Enabled': True}
-        for payload in username, pswd, roleid, enabled:
-            response = self.patch_request(uri, payload)
+    def add_user_via_patch(self, user):
+        response = self._find_account_uri(username=user.get('account_username'),
```

Comment:
I'm a little confused how this is supposed to work; shouldn't this try to find the "first empty slot" in the account list?

Reply:
The original behavior/design was that the slot to be used for the new account had to be specified based on the Id property. And I didn't change that.

But that would probably be a good enhancement. Is testing that the UserName property is an empty string a reasonable test for the empty slot?

Reply:
Yeah, I think that's a good enough check. I would imagine a normal call to add a user would simply specify the new user info and not give an ID or existing user name.

Reply:
Sounds good. I'll update the PR with that change.